src/main/java/org/springframework/data/util/TypeCollector.java

```diff
@@ -275,5 +334,111 @@ public boolean isEmpty() {
 		public int size() {
 			return mutableCache.size();
 		}
+
+	}
+
+	/**
+	 * Strategy interface providing predicates to filter types, fields, and methods from being introspected and
+	 * contributed to AOT processing.
+	 * <p>
+	 * {@code BeanRegistrationAotProcessor} implementations must be registered in a
+	 * {@value AotServices#FACTORIES_RESOURCE_LOCATION} resource. This interface serves as SPI and can be provided through
+	 * {@link org.springframework.beans.factory.aot.AotServices}.
+	 * <p>
+	 * {@link TypeCollector} discovers all implementations and applies the combined predicates returned by this interface
+	 * to filter unwanted reachable types from AOT contribution.
+	 *
+	 * @author Mark Paluch
+	 * @since 4.0
+	 */
+	public interface TypeCollectorPredicateProvider {
```

Comment:
What if we renamed this to `TypeCollectorPredicates`?

Reply:
a bit confusing, naming it predicates, `TypeCollectorFilterProvider` maybe?